miasm2/core/sembuilder.py

```diff
@@ -0,0 +1,155 @@
+"Helper to quickly build instruction's semantic side effects"
+
+import inspect
+import ast
+import re
+
+
+class MiasmTransformer(ast.NodeTransformer):
+    """AST visitor translating DSL to Miasm expression
+
+    memX[Y]       -> ExprMem(Y, X)
+    iX(Y)         -> ExprIntX(Y)
+    X if Y else Z -> ExprCond(Y, X, Z)
+    """
+
+    # Parsers
+    parse_integer = re.compile("^i([0-9]+)$")
+    parse_mem = re.compile("^mem([0-9]+)$")
+
+    # Visitors
+
+    def visit_Call(self, node):
+        """iX(Y) -> ExprIntX(Y)"""
+        # Match the function name
+        if not isinstance(node.func, ast.Name):
+            # TODO: launch visitor on node
+            return node
+
+        fc_name = node.func.id
+        new_name = fc_name
+        integer = self.parse_integer.findall(fc_name)
```

Comment:
Why do you use `.findall()` here? Since you only seem to need the first result, you may use `.search()` instead (not tested ;-)):

```python
    integer = self.parse_integer.search(fc_name)
    if integer is not None:
        new_name = "ExprInt%s" % integer.group()
```